lib/utils.js

```diff
@@ -4,3 +4,105 @@ export function PDFNumber(n) {
   // @see ISO 32000-1 Annex C.2 (real numbers)
   return Math.fround(n);
 }
+
+/**
+ * Measurement of size
+ *
+ * @typedef {number | `${number}` | `${number}${'em' | 'in' | 'px' | 'cm' | 'mm' | 'pc' | 'ex' | 'ch' | 'rem' | 'vw' | 'vmin' | 'vmax' | '%' | 'pt'}`} Size
+ */
+
+/**
+ * Measurement of how wide something is, false means 0 and true means 1
+ *
+ * @typedef {Size | boolean} Wideness
+ */
+
+/**
+ * Side definitions
+ * - To define all sides, use a single value
+ * - To define up-down left-right, use a `[Y, X]` array
+ * - To define each side, use `[top, right, bottom, left]` array
+ * - Or `{vertical: SideValue, horizontal: SideValue}`
+ * - Or `{top: SideValue, right: SideValue, bottom: SideValue, left: SideValue}`
+ *
+ * @template T
+ * @typedef {T| [T, T]| [T, T, T, T]| { vertical: T; horizontal: T }| { top: T; right: T; bottom: T; left: T }} SideDefinition<T>
+ **/
+
+/**
+ * @template T
+ * @typedef {{ top: T; right: T; bottom: T; left: T }} ExpandedSideDefinition<T>
+ */
+
+/**
+ * Convert any side definition into a static structure
+ *
+ * @template S
+ * @template D
+ * @template O
+ * @template {S | D} T
+ * @param {SideDefinition<S>} sides - The sides to convert
+ * @param {SideDefinition<D>} defaultDefinition - The value to use when no definition is provided
+ * @param {function(T): O} transformer - The transformation to apply to the sides once normalized
+ * @returns {ExpandedSideDefinition<O>}
+ */
+export function normalizeSides(
+  sides,
+  defaultDefinition = undefined,
+  transformer = (v) => v,
+) {
+  if (
+    sides === undefined ||
+    (typeof sides === "object" && Object.keys(sides).length === 0)
+  ) {
+    sides = defaultDefinition;
+  }
+  if (typeof sides !== "object" || sides === null || sides === undefined) {
```

Comment:
```suggestion
  if (typeof sides === "string" || typeof sides === "number" || !sides) {
```

Doing this check in the method start should allow to simplify further checks

Reply:
Added 👍